src/Data/Detection/DetectionRepository.cs

```diff
@@ -0,0 +1,36 @@
+namespace Data.Detection
+{
+    using System.Collections.Generic;
+    using Business.Detection.Common.Models;
+    using Business.Detection.Common.Repositories;
+
+    public class DetectionRepository : IDetectionRepository
+    {
+        private readonly Context _context;
+
+        public DetectionRepository(Context context)
+        {
+            _context = context;
+        }
+
+        public IEnumerable<MDetection> ByParameter()
+        {
+            var detections = _context.Set<EDetection>();
+
+            var detectionList = new List<MDetection>();
+            foreach (var detection in detections)
+            {
+                var detectionModel = new MDetection
+                {
+                    Id = detection.Id,
+                    Class = detection.Class,
+                    Score = detection.Score,
+                    Timestamp = detection.Timestamp,
+                };
+                detectionList.Add(detectionModel);
+            }
+
+            return detectionList;
```

Comment:
Alert, alert! C++ imperative style programming. Won't be easier to use `Linq` extensions instead?

Reply:
![image](https://user-images.githubusercontent.com/39530866/126166498-b0a29399-c2ba-4463-95db-09652da0848a.png)

Reply:
Done